src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/rewriting/impl/RuleAppliConvertTPAddIndexNLJToHashJoin.java

```diff
@@ -0,0 +1,36 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.rewriting.impl;
+
+import se.liu.ida.hefquin.engine.queryplan.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.LogicalOpJoin;
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.LogicalOpTPAdd;
+import se.liu.ida.hefquin.engine.queryplan.physical.BinaryPhysicalOp;
+import se.liu.ida.hefquin.engine.queryplan.physical.PhysicalOperatorForLogicalOperator;
+import se.liu.ida.hefquin.engine.queryplan.physical.impl.PhysicalOpHashJoin;
+import se.liu.ida.hefquin.engine.queryplan.physical.impl.PhysicalPlanWithBinaryRootImpl;
+import se.liu.ida.hefquin.engine.queryproc.impl.optimizer.rewriting.RewritingRule;
+import se.liu.ida.hefquin.engine.queryproc.impl.optimizer.rewriting.RuleApplicationBaseImpl;
+import se.liu.ida.hefquin.engine.queryproc.impl.optimizer.utils.ConstructSubPPBasedOnUnaryOperator;
+
+public class RuleAppliConvertTPAddIndexNLJToHashJoin extends RuleApplicationBaseImpl {
+    protected final ConstructSubPPBasedOnUnaryOperator helper = new ConstructSubPPBasedOnUnaryOperator();
+
+    public RuleAppliConvertTPAddIndexNLJToHashJoin( final PhysicalPlan[] subPlans ) {
+        super(subPlans);
+    }
+
+    @Override
+    public RewritingRule getRule() {
+        return new RuleConvertTPAddIndexNLJToHashJoin( 0.15 );
```

Comment:
No. The base class (RuleApplicationBaseImpl) should have a member for this, which is initialized by passing the rule in the constructor (i.e., in addition to passing `subPlan`).